pkg/controller/middleware/apikey.go

```diff
@@ -111,7 +109,7 @@ func RequireAPIKey(ctx context.Context, cacher cache.Cacher, db *database.Databa
 			// Save the authorized app on the context.
 			ctx = controller.WithAuthorizedApp(ctx, &authApp)
 			ctx = controller.WithRealm(ctx, &realm)
-			*r = *r.WithContext(ctx)
+			r = r.Clone(ctx)
```

Comment:
I think these should go back to r.WithContext since the deep copy isn't necessary